### PR TITLE
CI release workflow (+ deb package, homebrew formula)

### PR DESCRIPTION
**Pull request by coderobe**
Automatically push a draft release with binaries/packages for debian (deb) & macos (homebrew) on pushed tag  matching `v*`

Please squash when merging 🙈

### CODE REVIEW

**Comment by chevdor** [comment]
I kinda recognize the content so I don't have much to say. Did you test it ?
Especially, it requires a GH to be set if I recall, for the commit with the new Formula to work.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  create_draft:
+    needs: ["linux", "macos"]
+    name: Create Draft
+    runs-on: ubuntu-latest
+    outputs:
+      release_url: ${{ steps.create-release.outputs.html_url }}
+      asset_upload_url: ${{ steps.create-release.outputs.upload_url }}
+    steps:
+      - uses: actions/checkout@v2
```
Comment (chevdor) [inline .github/workflows/release.yml:142]: I think v3 is available. Also we are in a trend of currently pinning the gha using a ref. so I would suggest to pin with the ref of whatever is the current v3. Same for the others.

**Review by chevdor — CHANGES_REQUESTED** [review]
bump of the GHA versions + pinning to git ref.

**Comment by coderobe** [comment]
i did test it in a fork on my account. it works in the current state.

File: .github/workflows/release.yml (hunk 1 above)
Comment (coderobe) [reply .github/workflows/release.yml]: pinned all actions

**Review by chevdor — APPROVED** [review]

**Pull request merged**